src/dbnode/storage/namespace.go

```diff
@@ -1793,3 +1794,11 @@ func (n *dbNamespace) aggregateTiles(
 
 	return processedTileCount, nil
 }
+
+func (n *dbNamespace) DocRef(id ident.ID) (doc.Document, bool, error) {
```

Comment:
We've been sticking to the `type BytesID []byte` type alias everywhere in this whole flow so far, successfully.
Do I understand it correctly that passing it as an interface will cause allocation? Sadly, this might be difficult to avoid here, but still, something that I want to bring up.

Reply:
Yes that's true, you can use a re-useable ID from the caller if you like though:
https://godoc.org/github.com/m3db/m3/src/x/ident#NewReuseableBytesID
```go
var id ident.BytesID

bytesID := ident.NewReuseableBytesID() // Reuse this on consequent calls
bytesID.Reset(id)

d, ok, err := ns.DocRef(bytesID)
```

Reply:
Yea it will. I'm going to be pulling and putting the from the identifier pool.